File: sync/src/main/scala/com/fan/mysql/event/impl/RowDataEvent.java
package com.dingcloud.dts.binlog.event.impl;

import java.util.ArrayList;
import java.util.List;

import com.dingcloud.dts.binlog.event.AbstractEvent;
import com.dingcloud.dts.binlog.event.EventHeader;

public class RowDataEvent extends AbstractEvent {

	private long tableId;
	private List<RowData> rows;
	private int flags;

	public RowDataEvent(EventHeader header, long tableId) {
		super(header);
		this.tableId = tableId;
		this.rows = new ArrayList<RowData>(2);
	}

	public int getFlags() {
		return flags;
	}

	public void setFlags(int flags) {
		this.flags = flags;
	}

	public long getTableId() {
		return tableId;
	}

	public List<RowData> getRows() {
		return rows;
	}

}

File: async/src/main/scala/com/fan/mysql/async/binlog/event/impl/RowDataEvent.scala
package com.fan.mysql.async.binlog.event.impl

import java.util

import com.fan.mysql.async.binlog.event.{AbstractEvent, EventHeader}

class RowDataEvent(header: EventHeader, tableId: Long) extends AbstractEvent(header) {

  private val rows: util.List[RowData] = new util.ArrayList[RowData](2)
  private var flags = 0

  def getFlags: Int = flags

  def setFlags(flags: Int): Unit = {
    this.flags = flags
  }

  def getTableId: Long = tableId

  def getRows: util.List[RowData] = rows

}
